### ISSUE TITLE
Absolute links are generated for assets in the 404.html page

### ISSUE THREAD

**Issue by kcgthb** [squidfunk/mkdocs-material#335]
## Description

The 404 page doesn't seem to get the correct links for CSS/JS includes. 

### Expected behavior

Correct linking of external assets, using relative paths.

### Actual behavior

Wrong links are generated, using absolute paths. For instance, the link to modernizer.js is not the same in the `404.html` page an in others. 
* in `404.html`: `<script src="/mkdocs-material/assets/javascripts/modernizr.js"></script>`
* in other pages: `<script src="./assets/javascripts/modernizr.js"></script>`

### Steps to reproduce the bug

1. `git clone https://github.com/squidfunk/mkdocs-material.git`
2. `cd mkdocs-material && pip install -r requirements.txt`
3. `mkdocs serve`
4. Access `http://localhost:8000/not_found/` to generate a 404
5. See the `mkdocs` server logs:
   ```
   [I 170518 10:51:17 handlers:133] Browser Connected: http://localhost:8000/
   [W 170518 10:51:18 web:2063] 404 GET /not_found/ (127.0.0.1) 4.48ms
   [W 170518 10:51:18 web:2063] 404 GET /mkdocs-material/assets/javascripts/modernizr.js (127.0.0.1) 6.46ms
   [W 170518 10:51:18 web:2063] 404 GET /mkdocs-material/assets/javascripts/application.js (127.0.0.1) 10.44ms
   [W 170518 10:51:18 web:2063] 404 GET /mkdocs-material/assets/stylesheets/application.css (127.0.0.1) 9.01ms
   [W 170518 10:51:18 web:2063] 404 GET /mkdocs-material/assets/stylesheets/application.palette.css (127.0.0.1) 5.56ms
   [W 170518 10:51:18 web:2063] 404 GET /mkdocs-material/assets/javascripts/application.js (127.0.0.1) 3.35ms
   ```

### Package versions

* Python: `python --version`: `Python 2.7.13`
* MkDocs: `mkdocs --version`: `mkdocs, version 0.16.3`
* Material: `pip show mkdocs-material | grep -E ^Version`: `master`

### Project configuration

https://github.com/squidfunk/mkdocs-material/blob/master/mkdocs.yml

### System information

* OS: Debian 9.0
* Browser: Chrome 58

**Comment by squidfunk** [comment]
This is an old problem with MkDocs that is already reported on their issue tracker, see https://github.com/mkdocs/mkdocs/issues/77#issuecomment-242777289. 404 page handling is really bad currently. This is attributed to the fact that the `base_url` variable seems to default to the path in `site_url`, resulting in `/mkdocs-material`, and not `.`.

The issue is closed so from what I have read it seems that they fixed it. However, they scheduled the release of the fix for the 1.0.0 release, so we have to wait.

**Comment by kcgthb** [comment]
Oooh, ok, thanks for the clarification, I'll check out the upstream fix.
